src/components/NewMessageForm/NewMessageForm.vue

```diff
@@ -251,6 +281,25 @@ export default {
 
 			}
 		},
+		// Clicks the hidden file input when clicking the correspondent ActionButton, thus
+		// opening the file-picker
+		clickImportInput() {
+			this.$refs['file-upload-input'].click()
+		},
+		// Uploads the files to the root files directory
+		async processFiles(event) {
+			// Get the files array
+			const files = Object.values(event.target.files)
+			// Copy each file into the remote path
+			for (let i = 0; i < files.length; i++) {
+				const userId = this.$store.getters.getUserId()
+				try {
+					await client.putFileContents(`/files/${userId}/` + files[i].name, files[i], { overwrite: false })
```

Comment:
after upload, the path should be shared :laughing: 
So I guess it's time to split the API call from the filepicker in `handleFileShare` and reuse it in a shared function